playbooks/init/cluster_facts.yml

```diff
@@ -1,4 +1,49 @@
 ---
+# We need to ensure that users provide network variables that match an already
+# configured cluster.  We don't support changing these.
+- name: Read first master's config network settings
+  hosts: oo_first_master
+  gather_facts: no
+  tasks:
+  - stat:
+      path: "{{ openshift.common.config_base }}/master/master-config.yaml"
+    register: g_master_config_stat
+  - slurp:
+      src: "{{ openshift.common.config_base }}/master/master-config.yaml"
+    register: g_master_config_slurp
+  - name: Set existing cluster network setting facts
+    # These existing values will be checked against existing values in sanity_checks.
+    set_fact:
+      oo_first_master_osm_cluster_network_cidr: "{{ (g_master_config_slurp.content|b64decode|from_yaml).networkConfig.clusterNetworkCIDR }}"
+      oo_first_master_osm_host_subnet_length: "{{ (g_master_config_slurp.content|b64decode|from_yaml).networkConfig.hostSubnetLength }}"
+      oo_first_master_openshift_portal_net: "{{ (g_master_config_slurp.content|b64decode|from_yaml).networkConfig.serviceNetworkCIDR }}"
+    when: g_master_config_stat.stat.exists | bool
+
+  - name: Set first master openshift_portal_net
+    set_fact:
+      openshift_portal_net: "{{ oo_first_master_osm_cluster_network_cidr | default('10.128.0.0/14') }}"
+    when: openshift_portal_net is undefined
+
+  - name: Set first master osm_host_subnet_length
+    set_fact:
+      osm_host_subnet_length: "{{ oo_first_master_osm_host_subnet_length | default(9) }}"
+    when: osm_host_subnet_length is undefined
+
+  - name: Set first master osm_cluster_network_cidr
+    set_fact:
+      osm_cluster_network_cidr: "{{ oo_first_master_openshift_portal_net | default('172.30.0.0/16') }}"
+    when: osm_cluster_network_cidr is undefined
+
+- name: Set network facts for masters
```

Comment:
Alternatively, we could change all references to these variables in plays and roles to:
hostvars[groups.oo_first_master.0].openshift_portal_net
etc.